src/tiled/createobjecttool.h

```diff
@@ -66,6 +69,8 @@ class CreateObjectTool : public AbstractObjectTool
     ObjectGroupItem *mObjectGroupItem;
     MapObjectItem *mNewMapObjectItem;
     MapObjectItem *mOverlayPolygonItem;
+    bool mExtending;
+    bool mExtendingFirst;
```

Comment:
I tried overriding the functions `cancelNewMapObject()` and `finishNewMapObject()`, but apparently, the parent function gets called only and not the child one.